### PR TITLE
chore(golangci-lint): add gci linter and lint fix make target

### PR DESCRIPTION
**Pull request by sonasingh46**
Signed-off-by: Ashutosh Kumar <sonasingh46@gmail.com>

/kind cleanup

**What this PR does / why we need it**:
This PR adds `gci` linter 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Reference umbrella issue: https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1982

Additional Info: 

To fix lint error when you run `make-lint`, execute the following command: 
```bash
make lint-fix
```

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
None
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -297,7 +297,11 @@ $(YQ_BIN): $(YQ) ## Building yq from the tools folder
 
 .PHONY: lint
 lint: $(GOLANGCI_LINT) lint-latest ## Lint codebase
-	$(GOLANGCI_LINT) run -v
+	$(GOLANGCI_LINT) run -v $(GOLANGCI_LINT_EXTRA_ARGS)
+
+.PHONY: lint-fix
+lint-fix: $(GOLANGCI_LINT) ## Lint the codebase and run auto-fixers if supported by the linter
```
Comment (CecileRobertMichon) [inline Makefile:303]: could you please also add a note about this in https://capz.sigs.k8s.io/developers/development.html ?

Comment (sonasingh46) [reply Makefile:303]: Done. 
https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1987/files#diff-4d554fade8fefc047dcf883418cc72311cb5b62f3e0d20f7cbbfb849faa7335aR115

File: docs/book/src/developers/development.md [hunk 2]
```diff
@@ -112,6 +112,8 @@ Makefile targets and scripts are offered to work with go modules:
 - `make verify-modules` checks whether go module files are out of date.
 - `make modules` runs `go mod tidy` to ensure proper vendoring.
 - `hack/ensure-go.sh` checks that the Go version and environment variables are properly set.
+- `make lint` checks for the possible lint errors in the code base.
```
Comment (CecileRobertMichon) [inline docs/book/src/developers/development.md:115]: this section is specifically about go modules, not sure the lint targets belong in here

Comment (sonasingh46) [reply docs/book/src/developers/development.md]: Removed it.

**Comment by CecileRobertMichon** [comment]
opened https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2024 to fix to doc linter

**Comment by sonasingh46** [comment]
thanks @CecileRobertMichon

**Review by CecileRobertMichon — APPROVED** [review]
/lgtm

**Review by mboersma — APPROVED** [review]
/lgtm

**Pull request merged**